aiosysbus/api/nat.py

```diff
@@ -20,62 +32,82 @@ def set_firewall_PortForwarding(self, conf):
         """
         return self._access.post("Firewall", "setPortForwarding", conf)
 
-    def get_firewall_DMZ(self, conf=None):
+    def get_firewall_DMZ(
+        self, conf: dict[str, Any] | None = None
+    ) -> dict[str, Any] | None:
         """Get DMZ configuration"""
         return self._access.post("Firewall", "getDMZ", conf)
 
-    def set_firewall_DMZ(self, conf):
+    def set_firewall_DMZ(self, conf: dict[str, Any] | None) -> dict[str, Any] | None:
         """Set DMZ configuration"""
         return self._access.post("Firewall", "setDMZ", conf)
 
-    def set_firewall_IPv6Level(self, conf):
+    def set_firewall_IPv6Level(
+        self, conf: dict[str, Any] | None
+    ) -> dict[str, Any] | None:
         """Set level for IPv6 Firewall
 
         conf = {"parameters":{"level":"Low"}}
         """
         return self._access.post("Firewall", "setFirewallIPv6Level", conf)
 
-    def set_firewall_Level(self, conf):
+    def set_firewall_Level(self, conf: dict[str, Any] | None) -> dict[str, Any] | None:
         """Set level for IPv4 Firewall
 
         conf = {"parameters":{"level":"Low"}}
         """
         return self._access.post("Firewall", "setFirewallLevel", conf)
 
-    def set_firewall_Commit(self):
+    def set_firewall_Commit(self) -> dict[str, Any] | None:
         """Commit rules"""
         return self._access.post("Firewall", "commit")
 
-    def get_upnp_Pinhole(self, conf=None):
+    def get_upnp_Pinhole(
+        self, conf: dict[str, Any] | None = None
+    ) -> dict[str, Any] | None:
         """Upnp configuration
 
         conf = {"parameters":{"origin":"upnp"}}
         """
         return self._access.post("Firewall", "getPinhole", conf)
 
-    def get_dyndns_Hosts(self, conf=None):
+    def get_dyndns_Hosts(
+        self, conf: dict[str, Any] | None = None
+    ) -> dict[str, Any] | None:
         """Get Dynamic DNS Hosts configuration"""
         return self._access.post("DynDNS", "getHosts", conf)
 
-    def get_dyndns_Services(self, conf=None):
+    def get_dyndns_Services(
+        self, conf: dict[str, Any] | None = None
+    ) -> dict[str, Any] | None:
         """Get Dynamic DNS services configuration"""
         return self._access.post("DynDNS", "getServices", conf)
 
-    def set_dns_name(self, mac, conf):
+    def set_dns_name(
+        self, mac: str, conf: dict[str, Any] | None
+    ) -> dict[str, Any] | None:
         """Set DNS name
 
         mac = '00:01:2B:3C:4D:5E'
         conf = {"parameters":{"name":"nestor","source":"dns"}}
         """
-        return self._access.post("Devices/Device/" + mac + ":setName", conf)
-
-    def get_upnp_devices(self, conf=None):
+        return self._access.post(
+            "Devices/Device/" + mac + ":setName",
+            conf,  # type: ignore[arg-type]
+        )
```

Comment:
I think this is a bug, but it can maybe be fixed later.
For now I added `type: ignore` to keep existing behavior.

Reply:
yes indeed, there is a bug in these functions because a parameter is missing, they cannot work